### PR TITLE
noobaa cli enhancement - Create backingstore and namespacestore by specifying existing secret name

### PR DESCRIPTION
**Pull request by romayalon**
Signed-off-by: Romy <romy2232@gmail.com>

### CODE REVIEW

File: pkg/namespacestore/namespacestore.go [hunk 1]
```diff
@@ -265,11 +282,20 @@ func createCommon(cmd *cobra.Command, args []string, storeType nbv1.NSType, popu
 func RunCreateAWSS3(cmd *cobra.Command, args []string) {
 	createCommon(cmd, args, nbv1.NSStoreTypeAWSS3, func(namespaceStore *nbv1.NamespaceStore, secret *corev1.Secret) {
 		targetBucket := util.GetFlagStringOrPrompt(cmd, "target-bucket")
-		accessKey := util.GetFlagStringOrPromptPassword(cmd, "access-key")
-		secretKey := util.GetFlagStringOrPromptPassword(cmd, "secret-key")
 		region, _ := cmd.Flags().GetString("region")
-		secret.StringData["AWS_ACCESS_KEY_ID"] = accessKey
-		secret.StringData["AWS_SECRET_ACCESS_KEY"] = secretKey
+		secretName, _ := cmd.Flags().GetString("secret-name")
```
Comment (jeniawhite) [inline pkg/namespacestore/namespacestore.go:286]: Looks like this whole code can be in a function
The only difference is the keys, not the logic

File: pkg/backingstore/backingstore.go [hunk 2]
```diff
@@ -323,11 +344,19 @@ func createCommon(cmd *cobra.Command, args []string, storeType nbv1.StoreType, p
 func RunCreateAWSS3(cmd *cobra.Command, args []string) {
 	createCommon(cmd, args, nbv1.StoreTypeAWSS3, func(backStore *nbv1.BackingStore, secret *corev1.Secret) {
 		targetBucket := util.GetFlagStringOrPrompt(cmd, "target-bucket")
-		accessKey := util.GetFlagStringOrPromptPassword(cmd, "access-key")
-		secretKey := util.GetFlagStringOrPromptPassword(cmd, "secret-key")
 		region, _ := cmd.Flags().GetString("region")
-		secret.StringData["AWS_ACCESS_KEY_ID"] = accessKey
-		secret.StringData["AWS_SECRET_ACCESS_KEY"] = secretKey
+		secretName, _ := cmd.Flags().GetString("secret-name")
```
Comment (jeniawhite) [inline pkg/backingstore/backingstore.go:348]: Same thing as namespacestore

**Review by jeniawhite — APPROVED** [review]

File: pkg/namespacestore/namespacestore.go (hunk 1 above)
Comment (romayalon) [reply pkg/namespacestore/namespacestore.go:286]: let's keep it as it is so the flags will be handled at the same place and won't need to pass 4 keys for each call.

File: pkg/backingstore/backingstore.go (hunk 2 above)
Comment (romayalon) [reply pkg/backingstore/backingstore.go:348]: same

**Review by dannyzaken — APPROVED** [review]

**Pull request merged**